koocook_core/models/recipe.py

```diff
@@ -105,13 +103,14 @@ class Meta:
         unique_together = (('ip_address', 'user', 'recipe'), ('user', 'recipe'))
     ip_address = models.CharField(max_length=45)
     recipe = models.ForeignKey(Recipe, on_delete=models.CASCADE)
-    user = models.ForeignKey('koocook_core.KoocookUser', on_delete=models.SET_NULL, null=True)
+    user = models.ForeignKey('koocook_auth.KoocookUser', on_delete=models.SET_NULL, null=True)
     date_first_visited = models.DateTimeField(auto_now_add=True)
     date_last_visited = models.DateTimeField(auto_now=True)
 
     @classmethod
-    def associate_recipe_with_user(cls, user: 'koocook_core.KoocookUser', recipe: Recipe):
+    def associate_recipe_with_user(cls, user: 'koocook_auth.KoocookUser', recipe: Recipe):
         visit, created = cls.objects.get_or_create(user=user, recipe=recipe)
+        LOGGER.info(f"Incoming visit to {recipe.name} [{recipe.id}] from {user.name}")
```

Comment:
The log format here is different from the others. I suggest you find a way to standardise them. Maybe have each model have a log representation of itself, so that less work is done on the end. Logging visits could also be made a format, maybe using a function to get the log, so that the same format can be used with other pages, for example.

Reply:
This could be done later though.